semantics.md

```diff
@@ -1663,11 +1669,14 @@ group as determined by the group key value.
 
   - The argument must be statically typed to a numeric type.
 
-- COUNT - Counts the number of elements. COUNT(\*) counts all values
-  unconditionally,
-  COUNT([\<expression\>](#expressions)) counts all
-  values for which the expression does not result in NULL or
-  MISSING.
+- COUNT - Counts the number of elements. 
+  - COUNT(\*) counts all values unconditionally.
+  - COUNT([\<expression\>](#expressions)) counts all
+    values for which the expression does not result in NULL or
+    MISSING.
+  - COUNT(DISTINCT [\<expression\>](#expressions)) conditionally counts distinct values of the expression.
+  - COUNT(DISTINCT \*) unconditionally counts the number of distinct documents in the result set.
+  - COUNT(DISTINCT col1, col2, ...) counts distinct combinations of the specified columns.
```

Comment:
We should list the non-distinct version of this too.

Also could you note that `COUNT(DISTINCT? col1, col2, ..., coln)` is syntactically equivalent to `COUNT(DISTINCT? {'_1': col1, '_2': col2, ..., '_n': coln})`, but it is up to implementations whether or not to implement that as a syntactic rewrite. (That last bit is relevant since I did not include the rewrite as part of the spec tests! It isn't, strictly speaking, part of the spec to do the rewrite -- we just chose to implement it that way for convenience!)

Reply:
I added the non-distinct version and a note about the syntactic rewrite.